src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -368,9 +381,15 @@ export function createModelForService(
         }
     }
 
-    const usages = getUsages(program, convenienceOperations);
-    setUsage(usages, modelMap);
-    setUsage(usages, enumMap);
+    if (generateAllModels) {
+        navigateModels(program, serviceNamespaceType, modelMap, enumMap);
+        setUsageForAll(Usage.RoundTrip, modelMap);
+        setUsageForAll(Usage.RoundTrip, enumMap);
```

Comment:
Should we use the flag of `RoundTrip` for those unused models? I believe `None` should be sufficient. By definition, they are not round trip models, and the serialization/deserialization makes no sense for models like that, because our generated serialization/deserilaization only serves the payload.
If they want serialization and deserialization, they could add customized code to have those.

Reply:
No. If it is `None`, the model will not be generated. And customer want to generate all the models together with serialize and deserialize method. Because they want to use those model in client code.

Reply:
The first part is not correct. We still generate the models and enums when its usage is None. Changed in this PR: https://github.com/Azure/autorest.csharp/pull/3046
Since you already ensures the config for `unreferenced-types-handling` hard-coded to `keepAll`, those models will be there.

For second part, it is a consideration of breaking change when growing up. If we generate it as RoundTrip, in some version, if they change their cadl to involve it in an operation, suddenly its usage will be changed to Input or Output.
The usage cannot be changed to a smaller one (RoundTrip => Output will have breaking changes, but Output => RoundTrip will not have breaking changes)